examples/ref-templates/assets/htlc.py

```diff
@@ -0,0 +1,100 @@
+
+- If an argument arg_0 is passed to the script such that TMPL_HASHFN(arg_0) is equal to TMPL_HASHIMG, 
+then funds may be closed out to TMPL_RCV.
+- If txn.FirstValid is greater than TMPL_TIMEOUT, then funds may be closed out to TMPL_OWN.
+
+The idea is that by knowing the preimage to TMPL_HASHIMG, funds may be released to 
+TMPL_RCV (Scenario 1). Alternatively, after some timeout round TMPL_TIMEOUT, 
+funds may be closed back to their original owner, TMPL_OWN (Scenario 2). 
+Note that Scenario 1 may occur up until Scenario 2 occurs, even if TMPL_TIMEOUT has already passed."""
+
+""" Parameters
+TMPL_RCV: the address to send funds to when the preimage is supplied
+TMPL_HASHFN: the specific hash function (sha256 or keccak256) to use (sha256 in this example)
+TMPL_HASHIMG: the image of the hash function for which knowing the preimage under TMPL_HASHFN will release funds
+TMPL_TIMEOUT: the round after which funds may be closed out to TMPL_OWN
+TMPL_OWN: the address to refund funds to on timeout
+TMPL_FEE: maximum fee of any transactions approved by this contract """
+
+from pyteal import *
+
+john = Addr("2UBZKFR6RCZL7R24ZG327VKPTPJUPFM6WTG7PJG2ZJLU234F5RGXFLTAKA")
+master = Addr("WWYNX3TKQYVEREVSW6QQP3SXSFOCE3SKUSEIVJ7YAGUPEACNI5UGI4DZCE")
+fee = 10000
+pre_image = "QzYhq9JlYbn2QdOMrhyxVlNtNjeyvyJc/I8d8VAGfGc="
+timeout = 2000
+
+def htlc(TMPL_RCV = john,
+         TMPL_OWN = master,
+         TMPL_FEE = fee,
+         TMPL_HASHIMG = pre_image,
+         TMPL_HASHFN = Sha256,
+         TMPL_TIMEOUT = timeout):
+    
+    # First, check that the fee of this transaction is less than or equal to TMPL_FEE
+    fee_check = Txn.fee() < Int(TMPL_FEE)
+
+    # Next, check that this is a payment transaction. 
+    pay_check = Txn.type_enum() == TxnType.Payment
+
+    # Next, check that the Receiver field for this transaction is empty 
+    # Because this contract can approve transactions that close out its entire balance, 
+    # it should never have a receiver.
+    rec_field_check = Txn.receiver() == Global.zero_address()
+
+    # Next, check that the Amount of algos transferred is 0. This is for the same reason as 
+    # above: we only allow transactions that close out this account completely, which 
+    # having a non-zero-address CloseRemainderTo will handle for us.
+    amount_check = Txn.amount() == Int(0)
+
+    # fold all the above checks into a single boolean.
+    required_condition = And(
+        fee_check,
+        pay_check,
+        rec_field_check,
+        amount_check
+    )
+
+    """Payout scenarios : At this point in the execution, there is one boolean variable on the 
+    stack that must be true in order for the transaction to be valid. The checks we have done 
+    above apply to any transaction that may be approved by this script.We will now check if we
+     are in one of the two payment scenarios described in the functionality section."""
```

Comment:
this is a doc string, it doesn't make sens to use it here. We use it only to function, classes, globals etc...
Let's use normal coment:
```
# 
# 
```

Reply:
ok